src/rocksdb/src/lib.rs

```diff
@@ -869,7 +855,10 @@ fn rocksdb_core_loop<K, V, M, O, IM, F>(
                             }
                             Err(e) => match e.kind() {
                                 ErrorKind::TryAgain => RetryResult::RetryableErr(Error::RocksDB(e)),
-                                _ => RetryResult::FatalErr(Error::RocksDB(e)),
+                                _ => {
+                                    fatal_error.replace(Error::RocksDB(e.clone()));
```

Comment:
nit: `Option::replace` is mostly useful when you want to do something with the old value, otherwise this code is the same as the normal assignment `fatal_error = Some(..)`.